include/StreamBase.h

```diff
@@ -0,0 +1,230 @@
+        /// @brief Gets the name of the AssetGroup this Stream is streaming to/from.
+        /// @remarks The returned Stream can be empty, which usually means it wasn't created by the Resource system.
+        /// @return Returns a String containing the identifier this Stream was grouped with.
+        [[nodiscard]] virtual String GetGroup() const = 0;
+
+        /// @brief Gets the size of the Stream.
+        /// @return Returns the size of this Stream in bytes.
+        [[nodiscard]] virtual StreamSize GetSize() const = 0;
+        /// @brief Gets whether or not this Stream supports seeking.
+        /// @return Returns true if this Stream supports seeking to any point in the Stream, false otherwise.
+        [[nodiscard]] virtual Boole CanSeek() const = 0;
+        /// @brief Gets whether or not the source of this Stream is encrypted.
+        /// @return Returns true if the original data being streamed is encrypted, false otherwise.
+        [[nodiscard]] virtual Boole IsEncrypted() const = 0;
+        /// @brief Gets whether or not this Stream is decompressing/decrypting the source data.
+        /// @remarks A Raw Stream is fairly common for file system and memory streams.  Even with Zip or other
+        /// similar archive streams it can be useful if custom operations are desired, but less common.
+        /// @return Returns true if the source data is being streamed without being altered, false otherwise.
+        [[nodiscard]] virtual Boole IsRaw() const = 0;
+
+        /// @brief Gets whether or not the current position is at the end of the Stream.
+        /// @return Returns true if the current position has reached the end of the Stream, false otherwise.
+        [[nodiscard]] virtual Boole EoF() const = 0;
+        /// @brief Gets whether or not a critical error was detected in a previous operation in the Stream.
+        /// @return Returns true if a critical error has occurred and the Stream integrity may be invalid, false otherwise.
+        [[nodiscard]] virtual Boole Bad() const = 0;
+        /// @brief Gets whether or not an otherwise silent and recoverable error was detected in a previous operation in the Stream.
+        /// @return Returns true if a non-critical operation failed (such as seek), false otherwise.
+        [[nodiscard]] virtual Boole Fail() const = 0;
+        /// @brief Gets whether or not this Stream is intact and ready for operations.
+        /// @return Returns true if no failures have been detected, false otherwise.
+        [[nodiscard]] virtual Boole IsValid() const = 0;
+
+        /// @brief Clears any stored error state on the Stream.
+        /// @remarks This is useful for non-critical errors such as ones that cause "EoF()" or "Fail()" to return true but not
+        /// "Bad()".  Using this to clear critical errors is not advised.
+        virtual void ClearErrors() = 0;
+    };//iStreamBase
+
+    #ifdef _MSC_VER
+    #pragma vtordisp(push, 2)
```

Comment:
What is this? A comment is probably reasonable.

Reply:
A comment was added.